stdlib/cmake/modules/AddSwiftStdlib.cmake

```diff
@@ -1346,6 +1346,18 @@ function(add_swift_target_library_single target name)
       target_link_options(${target} PRIVATE
         "LINKER:-current_version,${SWIFT_COMPILER_VERSION}")
     endif()
+
+    # In the past, we relied on unsetting globally
+    # CMAKE_OSX_ARCHITECTURES to ensure that CMake would
+    # not add the -arch flag
+    # This is no longer the case when running on Apple Silicon,
+    # when CMake will enforce a default (see
+    # https://gitlab.kitware.com/cmake/cmake/-/merge_requests/5291)
+    set_property(TARGET ${target} PROPERTY OSX_ARCHITECTURES "${SWIFTLIB_SINGLE_ARCHITECTURE}")
+    if (target_static)
```

Comment:
Should this perhaps be `if(TARGET ${target_static})`?  I'm worried that the string may be non-empty, which would result in this evaluating to true, even though the target has not been setup.

Reply:
Good point -- implemented in 848a488b5d9